charts/signoz/templates/query-service/statefulset.yaml

```
@@ -140,7 +140,16 @@ spec:
               value: {{ .Values.queryService.configVars.deploymentType }}
             {{- range $key, $val := .Values.queryService.additionalEnvs }}
             - name: {{ $key }}
-              value: {{ $val | toYaml }}
+              {{- if kindIs "map" $val }}
+              {{- if $val.value}}
+              value: {{ $val.value | quote }}
+              {{- end}}
+              {{- if $val.valueFrom }}
+              valueFrom: {{ toYaml $val.valueFrom | nindent 16 }}
+              {{- end}}
+              {{- else }}
+              value: {{ $val | quote }}
+              {{- end}}
```

@TheShubhendra, 

- Can you make a define block out of this in `_helpers.tpl`? Once you create the block in `_helpers.tpl`, we can use it here.
- We need to ensure that all keys are uppercased and duplicate keys are not present. 

Something like the snippet below. It takes the first value if multiple values for a key are present. You just need to call `upper` function when setting the names.

```suggestion
{{- define "queryService.renderAdditionalEnv" -}}

{{- $dict := . -}}

{{- range keys . | sortAlpha }}
{{- $val := pluck . $dict | first -}}
{{- $valueType := printf "%T" $val -}}
{{ if eq $valueType "map[string]interface {}" }}
- name: {{ . }}
{{ toYaml $val | indent 2 -}}
{{- else if eq $valueType "string" }}
{{- if regexMatch "valueFrom" $val }}
- name: {{ . }}
{{ $val | indent 2 }}
{{- else }}
- name: {{ . }}
  value: {{ $val | quote }}
{{- end }}
{{- else }}
- name: {{ . }}
  value: {{ $val | quote }}
{{- end }}
{{- end -}}

{{- end -}}
```

---

I have created a common function, can you please check? Thanks!